[36746] Unset plain password accessor after saving

oliverguenther opened the pull request:
Otherwise, this password remains and is validated again due to
"user.register!" which then fails with an exception

https://community.openproject.org/wp/36746

oliverguenther committed:
[36746] Unset plain password accessor after saving

Otherwise, this password remains and is validated again due to
"user.register!" which then fails with an exception

https://community.openproject.org/wp/36746

machisuji reviewed (approved):
👍

machisuji commented:
@oliverguenther Some specs* seem to require an update after this fix.

* e.g. `rspec ./spec_legacy/functional/users_controller_spec.rb:210 # UsersController should update with password change should send a notification`

oliverguenther committed:
Don't try to use user.password after saving

it now gets cleared

oliverguenther committed:
Remove user.password usage in specs

The user password attribute is an accessor that now gets deleted after
saving so tests can no longer depend on it being present after created
from a Factory

oliverguenther committed:
Also retain user password in mail handler

oliverguenther commented:
@machisuji  We can now no longer rely on a user password being present after `#save`. This was used in some tests and the mail handler where a user is created on the fly and is now corrected
